File: OpenLive-iOS-Objective-C/OpenLiveUITests/OpenLiveUITests.m
//
//  OpenLiveUITests.m
//  OpenLiveUITests
//
//  Created by GongYuhua on 2017/1/13.
//  Copyright © 2017年 Agora. All rights reserved.
//

#import <XCTest/XCTest.h>

@interface OpenLiveUITests : XCTestCase

@end

@implementation OpenLiveUITests

- (void)setUp {
    [super setUp];
    
    self.continueAfterFailure = NO;
    [[[XCUIApplication alloc] init] launch];
}

- (void)tearDown {
    [super tearDown];
}

- (void)testJoinAndLeaveChannel {
    XCUIApplication *app = [[XCUIApplication alloc] init];
    XCUIElement *inputNameOfLiveTextField = app.textFields[@"Input name of Live"];
    [inputNameOfLiveTextField tap];
    [inputNameOfLiveTextField typeText:@"uiTestChannel\n"];
    [app.sheets.buttons[@"Broadcaster"] tap];
    
    XCUIElement *closeButton = app.buttons[@"btn close"];
    [self expectationForPredicate:[NSPredicate predicateWithFormat:@"exists == 1"] evaluatedWithObject:closeButton handler:nil];
    [self waitForExpectationsWithTimeout:5 handler:nil];
    
    [closeButton tap];
    
    XCUIElement *textField = app.textFields[@"Input name of Live"];
    [self expectationForPredicate:[NSPredicate predicateWithFormat:@"exists == 1"] evaluatedWithObject:textField handler:nil];
    [self waitForExpectationsWithTimeout:5 handler:nil];
}

@end

File: OpenLive-iOS/OpenLiveUITests/OpenLiveUITests.swift
//
//  OpenLiveUITests.swift
//  OpenLiveUITests
//
//  Created by GongYuhua on 2017/1/12.
//  Copyright © 2017年 Agora. All rights reserved.
//

import XCTest

class OpenLiveUITests: XCTestCase {
    
    override func setUp() {
        super.setUp()
        
        continueAfterFailure = false
        XCUIApplication().launch()
    }
    
    override func tearDown() {
        super.tearDown()
    }
    
    func testJoinAndLeaveChannel() {
        
        let app = XCUIApplication()
        let inputNameOfLiveTextField = app.textFields["Input name of Live"]
        inputNameOfLiveTextField.tap()
        inputNameOfLiveTextField.typeText("uiTestChannel\n")
        app.sheets.buttons["Broadcaster"].tap()
        
        let closeButton = app.buttons["btn close"]
        expectation(for: NSPredicate(format: "exists == 1"), evaluatedWith: closeButton, handler: nil)
        waitForExpectations(timeout: 5, handler: nil)
        
        closeButton.tap()
        
        let textField = app.textFields["Input name of Live"]
        expectation(for: NSPredicate(format: "exists == 1"), evaluatedWith: textField, handler: nil)
        waitForExpectations(timeout: 5, handler: nil)
    }
}
